nightly-job/sanitation_job.py

```diff
@@ -25,44 +25,47 @@ async def run_sanitation(args):
 
     try:
         total_run = 0
+        total_allow_listed = 0
         total_deemed_sanitary = 0
         summary_run_data = {}
         summary_language_data = {}
         yesterday = datetime.utcnow().date() - timedelta(days=1)
-        
+    
         data_validation_sample = pd.DataFrame()
-
-        unsanitized_search_term_stream = stream_search_terms()  # load unsanitized search terms
-        
+    
+        unsanitized_search_term_stream = stream_search_terms() # load unsanitized search terms
         for raw_page in unsanitized_search_term_stream:
             total_run += raw_page.shape[0]
-            
+        
             one_percent_sample = raw_page.sample(frac = 0.01)
             data_validation_sample = data_validation_sample.append(one_percent_sample)
-            
-            pii_in_query_mask, run_data, language_data = await detect_pii(raw_page['query'], census_surnames)
-            sanitized_page = raw_page.loc[
-                ~numpy.array(pii_in_query_mask)]  # ~ reverses the mask so we get the queries WITHOUT PII in them
-            total_deemed_sanitary += sanitized_page.shape[0]
+        
+            allow_listed_terms_page = raw_page.loc[raw_page.present_in_allow_list]
+            unsanitized_unallowlisted_terms = raw_page.loc[~raw_page.present_in_allow_list]
 
+            pii_in_query_mask, run_data, language_data = await detect_pii(unsanitized_unallowlisted_terms['query'], census_surnames)
+            sanitized_page = unsanitized_unallowlisted_terms.loc[~numpy.array(pii_in_query_mask)] # ~ reverses the mask so we get the queries WITHOUT PII in them
+            total_allow_listed += allow_listed_terms_page.shape[0]
+            total_deemed_sanitary += (sanitized_page.shape[0] + total_allow_listed)
```

Comment:
Should this be `allow_listed_terms_page.shape[0]` that gets added here?

Reply:
1. @dzeber you are right
2. You pointing this out made me realize that the way this is written is circuitous and a little confusing
3. I will fix it posthaste
4. Thank you!!

Reply:
@dzeber I put in the fix and moved some lines of code around for better legibility. I am running the script now and will check the numbers when it finishes; it might not finish before 5 PM so this is on my docket for first thing in the morning. Will let you know once the numbers look good and this is ready for another pass!